index.js

```diff
@@ -373,6 +393,88 @@ function sanitizeHtml(html, options, _recursing) {
     return !options.allowedSchemes || options.allowedSchemes.indexOf(scheme) === -1;
   }
 
+/**
+   * Filters user input css properties by whitelisted regex attributes.
+   *
+   * @param {object} abstractSyntaxTree  - Object representation of CSS attributes.
+   * @property {array[Declaration]} abstractSyntaxTree.nodes[0] - Each object cointains prop and value key, i.e { prop: 'color', value: 'red' }.
+   * @param {object} allowedStyles       - Keys are properties (i.e color), value is list of permitted regex rules (i.e /green/i).
+   * @return {object}                    - Abstract Syntax Tree with filtered style attributes.
+   */
+  function filterCss(abstractSyntaxTree, allowedStyles) {
+    if (!allowedStyles) {
+      return abstractSyntaxTree;
+    }
+
+    var filteredAST = cloneDeep(abstractSyntaxTree);
+    var astRules = abstractSyntaxTree.nodes[0];
+    var selectedRule;
+
+    // Merge global and tag-specific styles into new AST.
+    if (allowedStyles[astRules.selector] && allowedStyles['*']) {
+      selectedRule = mergeWith(
+        cloneDeep(allowedStyles[astRules.selector]),
+        allowedStyles['*'],
+        function(objValue, srcValue) {
+          if (isArray(objValue)) {
+            return objValue.concat(srcValue);
+          }
+        }
+      );
+    } else {
+      selectedRule = allowedStyles[astRules.selector] || allowedStyles['*'];
+    }
+
+    /**
+     * Filters the existing attributes for the given property. Discards any attributes
```

Comment:
Did you mean to include this for the `astrules.nodes.reduce` callback?

Function docblocks shouldn't be inline like this. It probably makes sense to move the callback out of this block and make it a named function.

Reply:
Yeah I did. It's complex enough to warrant being moved out.